docs/reference/hooks.rst

```diff
@@ -976,3 +975,51 @@ Document serving
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
   Called when Wagtail is about to serve a document. The callable passed into the hook will receive the document object and the request object. If the callable returns an ``HttpResponse``, that response will be returned immediately to the user, instead of serving the document. Note that this hook will be skipped if the :ref:`WAGTAILDOCS_SERVE_METHOD <wagtaildocs_serve_method>` setting is set to ``direct``.
+
+
+Snippets
+--------
+
+Hooks for working with registered Snippets.
+
+.. _after_edit_snippet:
+
+``after_edit_snippet``
+~~~~~~~~~~~~~~~~~~~~~~
+
+  Called when a Snippet is edited. The callable passed into the hook will receive the model instance, the request object. If the callable returns an ``HttpResponse``, that response will be returned immediately to the user, and Wagtail will not proceed to call ``redirect()`` to the listing view.
+
+  .. code-block:: python
+
+    from wagtail.core import hooks
+
+    @hooks.register('after_edit_snippet')
+    def after_snippet_update(request, instance):
+        # your code here
+        # print(instance.pk)
+
+.. _after_create_snippet:
+
+``after_create_snippet``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+  Called when a Snippet is created. ``after_create_snippet`` and
+  ``after_edit_snippet`` work in identical ways. The only difference is where
+  the hook is called.
+
+.. _after_delete_snippet:
+
+``after_delete_snippet``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+  Called when a Snippet is deleted. The callable passed into the hook will receive the model instance(s) as a queryset along with the request object. If the callable returns an ``HttpResponse``, that response will be returned immediately to the user, and Wagtail will not proceed to call ``redirect()`` to the listing view.
+
+  .. code-block:: python
+
+    from wagtail.core import hooks
+
+    @hooks.register('after_delete_snippet')
+    def after_snippet_delete(request, instances):
+        # "instances" is a QuerySet
+        for instance in instances:
+            print(instance.pk)
```

Comment:
The page CRUD hooks use `HttpResponse` as an example. Might be worth doing that instead of one the print statements to have a more realistic code example?

Reply:
e.g.
```
from django.http import HttpResponse
...
total = len(instances)
return HttpResponse(f"{total} have been deleted", content_type="text/plain")
```